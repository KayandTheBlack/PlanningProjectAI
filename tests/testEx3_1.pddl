(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI - city
    hotelA_A hotelA_B hotelA_C hotelB_A hotelB_B hotelB_C hotelC_A hotelC_B hotelC_C hotelD_A hotelD_B hotelD_C hotelE_A hotelE_B hotelE_C hotelF_A hotelF_B hotelF_C hotelG_A hotelG_B hotelG_C hotelH_A hotelH_B hotelH_C hotelI_A hotelI_B hotelI_C - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelA_C cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelB_C cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelC_C cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelD_C cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelE_C cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelF_C cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelG_C cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelH_C cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelI_C cityI)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight cityA cityE)
    (flight cityA cityF)
    (flight cityA cityI)
    (flight cityB cityD)
    (flight cityB cityE)
    (flight cityB cityF)
    (flight cityC cityE)
    (flight cityC cityF)
    (flight cityC cityH)
    (flight cityD cityC)
    (flight cityD cityG)
    (flight cityD cityH)
    (flight cityE cityB)
    (flight cityE cityD)
    (flight cityE cityG)
    (flight cityF cityA)
    (flight cityF cityB)
    (flight cityF cityH)
    (flight cityG cityA)
    (flight cityG cityD)
    (flight cityG cityI)
    (flight cityH cityA)
    (flight cityH cityD)
    (flight cityH cityF)
    (flight cityI cityF)
    (flight cityI cityG)
    (flight cityI cityH)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 2)
    (= (maxDaysInCity) 3)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (totalcost) 0)
    (= (hotelcost hotelA_A) 207)
    (= (hotelcost hotelA_B) 161)
    (= (hotelcost hotelA_C) 158)
    (= (hotelcost hotelB_A) 134)
    (= (hotelcost hotelB_B) 109)
    (= (hotelcost hotelB_C) 161)
    (= (hotelcost hotelC_A) 161)
    (= (hotelcost hotelC_B) 129)
    (= (hotelcost hotelC_C) 191)
    (= (hotelcost hotelD_A) 138)
    (= (hotelcost hotelD_B) 83)
    (= (hotelcost hotelD_C) 229)
    (= (hotelcost hotelE_A) 110)
    (= (hotelcost hotelE_B) 87)
    (= (hotelcost hotelE_C) 156)
    (= (hotelcost hotelF_A) 227)
    (= (hotelcost hotelF_B) 119)
    (= (hotelcost hotelF_C) 88)
    (= (hotelcost hotelG_A) 170)
    (= (hotelcost hotelG_B) 54)
    (= (hotelcost hotelG_C) 122)
    (= (hotelcost hotelH_A) 194)
    (= (hotelcost hotelH_B) 113)
    (= (hotelcost hotelH_C) 232)
    (= (hotelcost hotelI_A) 181)
    (= (hotelcost hotelI_B) 228)
    (= (hotelcost hotelI_C) 202)

    (= (flightcost Origin cityA) 0)
    (= (flightcost Origin cityB) 0)
    (= (flightcost Origin cityC) 0)
    (= (flightcost Origin cityD) 0)
    (= (flightcost Origin cityE) 0)
    (= (flightcost Origin cityF) 0)
    (= (flightcost Origin cityG) 0)
    (= (flightcost Origin cityH) 0)
    (= (flightcost Origin cityI) 0)
    (= (flightcost cityA cityE) 184)
    (= (flightcost cityA cityF) 153)
    (= (flightcost cityA cityI) 147)
    (= (flightcost cityB cityD) 151)
    (= (flightcost cityB cityE) 110)
    (= (flightcost cityB cityF) 110)
    (= (flightcost cityC cityE) 112)
    (= (flightcost cityC cityF) 147)
    (= (flightcost cityC cityH) 169)
    (= (flightcost cityD cityC) 123)
    (= (flightcost cityD cityG) 158)
    (= (flightcost cityD cityH) 101)
    (= (flightcost cityE cityB) 164)
    (= (flightcost cityE cityD) 198)
    (= (flightcost cityE cityG) 186)
    (= (flightcost cityF cityA) 195)
    (= (flightcost cityF cityB) 110)
    (= (flightcost cityF cityH) 123)
    (= (flightcost cityG cityA) 102)
    (= (flightcost cityG cityD) 139)
    (= (flightcost cityG cityI) 145)
    (= (flightcost cityH cityA) 192)
    (= (flightcost cityH cityD) 159)
    (= (flightcost cityH cityF) 101)
    (= (flightcost cityI cityF) 165)
    (= (flightcost cityI cityG) 156)
    (= (flightcost cityI cityH) 116)

  )

  (:goal
    (and
      (>= (citiesVisited) 7)
      (>= (totalDays) 10)
      (>= (totalcost) 2000)
      (<= (totalcost) 3000)
    )
  )

  (:metric 
    minimize (totalcost)
  )

)
