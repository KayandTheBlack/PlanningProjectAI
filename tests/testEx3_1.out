
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((1.00*[RF0](TOTALCOST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((1.00*[RF0](TOTALCOST)) - () + 0.00)

advancing to distance:    7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYB HOTELB_B
        1: FLYANDSTAY CITYB CITYE HOTELE_B
        2: FLYANDSTAY CITYE CITYD HOTELD_A
        3: FLYANDSTAY CITYD CITYG HOTELG_A
        4: FLYANDSTAY CITYG CITYA HOTELA_C
        5: FLYANDSTAY CITYA CITYF HOTELF_C
        6: FLYANDSTAY CITYF CITYH HOTELH_B
     

time spent:    0.00 seconds instantiating 135 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 56 facts and 135 actions
               0.00 seconds creating final representation with 55 relevant facts, 23 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.00 seconds searching, evaluating 88 states, to a max depth of 0
               0.00 seconds total time

